test/parallel/test-util-inherits.js

```diff
@@ -40,6 +40,40 @@ const c = new C();
 assert.strictEqual(c.getValue(), 'abc');
 assert.strictEqual(c.constructor, C);
 
+// inherits can be called after setting prototype properties
+function D() {
+  C.call(this);
+  this._d = 'd';
+}
+
+D.prototype.d = function() { return this._d; };
+inherits(D, C);
+
+assert.strictEqual(D.super_, C);
+
+const d = new D();
+assert.strictEqual(d.c(), 'c');
+assert.strictEqual(d.d(), 'd');
+assert.strictEqual(d.constructor, D);
+
+// ES6 classes can inherit from a constructor function
+class E {
+  constructor() {
+    D.call(this);
```

Comment:
Would this be better written as `E.super_.call(this)` ? That decouples E's constructor from the inherits in ln67.